saleor/graphql/order/filters.py

```diff
@@ -26,6 +28,13 @@ def get_payment_id_from_query(value):
         return None
 
 
+def get_order_id_from_query(value):
+    if value.startswith("#"):
+        value = value[1:]
+    if value.isnumeric():
+        return value
```

Comment:
Maybe `return value if value.isnumeric() else None` ? Personally I prefer to see that None can also be returned explicitly but maybe that's just my habit.